pkg/kubelet/dockertools/manager.go

```diff
@@ -216,13 +218,21 @@ func NewDockerManager(
 		cpuCFSQuota:            cpuCFSQuota,
 	}
 	dm.runner = lifecycle.NewHandlerRunner(httpClient, dm, dm)
-	if serializeImagePulls {
-		dm.imagePuller = kubecontainer.NewSerializedImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
-	} else {
-		dm.imagePuller = kubecontainer.NewImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
-	}
 	dm.containerGC = NewContainerGC(client, containerLogsDir)
 
+	if dm.isDockerAPIVersionCompatible(minimumDockerAPIVersionForParallelPulls) {
```

Comment:
I was hoping this entire if/else would look like this:

```go
if dm.shouldSerializeImagePulls(minimumDockerAPIVersionForParallelPulls, serializeImagePulls) {
  dm.imagePuller = kubecontainer.NewSerializedImagePuller ...
} else {
  dm.imagePuller = kubecontainer.NewImagePuller ...
}
```

We want to avoid specifying NewImagePuller and NewSerializedImagePuller multiple times